contrib/auto-render/auto-render.js

```diff
@@ -25,10 +25,11 @@ const renderMathInText = function(text, delimiters) {
         } else {
             const span = document.createElement("span");
             const math = data[i].data;
+            // Override any display mode defined in the settings with that
+            // defined by the text itself
+            renderOptions.displayMode = data[i].display;
```

Comment:
Ah, I see now -- the copy is done by the caller.  Maybe rename `renderOptions` to `optionsCopy` and add a comment to note that it's modified within the function, in case this API is ever exposed.